DigitalEcosystem/exploratory/matproj_parent_eAboveHull/e_above_hull.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[]:
+
+
+import copy
+import os
+import collections
+
+import pandas as pd
+import numpy as np
+import optuna
+import xgboost
+import imblearn.over_sampling
+import sklearn.model_selection
+import dscribe.descriptors
+import tqdm
+import sklearn.pipeline
+import pymatgen.ext.matproj
+import functools
+
+import matplotlib.pyplot as plt
+import sklearn.impute
+import seaborn as sns
+
+import sys
+sys.path.append("../../../")
+import DigitalEcosystem.utils.figures
+import DigitalEcosystem.utils.misc
+import DigitalEcosystem.utils.functional
+
+tqdm.tqdm.pandas()
+
+
+# In[]:
+
+
+# Random seeds for reproducibility
+RANDOM_SEED = 1234
+import random
+random.seed(RANDOM_SEED)
+np.random.seed(RANDOM_SEED)
+
+
+# In[]:
+
+
+# Load up the data
+data_path = "../../refined/httpot/full_featurized_data.pkl"
+data = pd.read_pickle(data_path)
+
+data = data[data['discovery_process (unitless)']=='top-down']
+
+cols_to_drop = ['formula',
+                'discovery_process (unitless)',
+                'potcars (unitless)',
+                'is_hubbard (unitless)',
+                'energy_per_atom (eV)',
+                'exfoliation_energy_per_atom (eV/atom)',
+                'is_bandgap_direct (unitless)',
+                'is_metal (unitless)',
+                'energy_vdw_per_atom (eV/atom)',
+                'total_magnetization (Bohr Magneton)']
+target_column = ['bandgap (eV)']
+matpedia_id = ['2dm_id (unitless)']
+atoms_col = ['atoms_object (unitless)']
```

Comment:
Same things as before:

- shortnames should be avoided: `decomp` -> `decomposition` etc.
- no explanations in this notebook

Reply:
Pasted these into a `misc` module and added comments

Reply:
Made those changes, and added notes